official/utils/logging/hooks.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import examples_per_second_hook
+
+def get_train_hooks(names):
+  """Factory for getting a list of TensorFlow hooks for training by name.
+
+  Args:
+    names: case-insensitive, comma-separated string with names of desired hook
+      classes. Allowed: LoggingTensorHook, ProfilerHook, ExamplesPerSecondHook
+
+  Returns:
+    list of instantiated hooks, ready to be used in a classifier.train call
+
+  Raises:
+    ValueError: if an unrecognized name is passed
+  """
+
+  if not names:
+    raise ValueError('Hook names cannot be None/Empty')
```

Comment:
I think it should be able to be empty-- we should just return an empty list. Better to just check `if names is None`?